BDArmory/Control/BDModuleOrbitalAI.cs

```diff
@@ -485,15 +485,16 @@ void Maneuver()
                     {
                         Orbit o = vessel.orbit;
                         double UT = Planetarium.GetUniversalTime();
-                        if (!ongoingOrbitCorrection && (o.ApA < 0 && o.timeToPe < -60))
+                        var descending = o.timeToPe > 0 && o.timeToPe < o.timeToAp;
+                        if ((!ongoingOrbitCorrection && (o.ApA < 0 && o.timeToPe < -60)) || (ongoingOrbitCorrection && (o.ApA / o.PeA > 10f)))
```

Comment:
This condition isn't right. When the correction is ongoing, the orbit can still have `o.ApA < 0 && o.timeToPe < -60`, causing the craft to switch between "correcting orbit" and "manuevering to intercept". The `ongoingOrbitCorrection` boolean isn't really sufficient by itself to distinguish between the reasons for correcting the orbit, so I think adding something like the following should help:
```diff
diff --git a/BDArmory/Control/BDModuleOrbitalAI.cs b/BDArmory/Control/BDModuleOrbitalAI.cs
index 31b837a6d..89d163b57 100644
--- a/BDArmory/Control/BDModuleOrbitalAI.cs
+++ b/BDArmory/Control/BDModuleOrbitalAI.cs
@@ -42,3 +42,4 @@ public class BDModuleOrbitalAI : BDGenericAIBase, IBDAIControl
         private bool maneuverStateChanged = false;
-        private bool ongoingOrbitCorrection = false;
+        enum OrbitCorrectionReason { None, FallingInsideAtmosphere, ApoapsisLow, PeriapsisLow, Escaping };
+        private OrbitCorrectionReason ongoingOrbitCorrectionDueTo = OrbitCorrectionReason.None;
         private bool wasDescendingUnsafe = false;
@@ -488,3 +489,6 @@ void Maneuver()
                         var descending = o.timeToPe > 0 && o.timeToPe < o.timeToAp;
-                        if ((!ongoingOrbitCorrection && (o.ApA < 0 && o.timeToPe < -60)) || (ongoingOrbitCorrection && (o.ApA / o.PeA > 10f)))
+                        if (o.altitude > minSafeAltitude && (
+                            (ongoingOrbitCorrectionDueTo == OrbitCorrectionReason.None && o.ApA < 0 && o.timeToPe < -60) ||
+                            (ongoingOrbitCorrectionDueTo == OrbitCorrectionReason.Escaping && ((o.ApA < 0 && o.timeToPe < -60) || (o.ApA / o.PeA > 10f)))
+                        ))
                         {
@@ -492,3 +496,4 @@ void Maneuver()
                             SetStatus("Correcting Orbit (On escape trajectory)");
-                            ongoingOrbitCorrection = o.ApA / o.PeA > 10f;
+                            ongoingOrbitCorrectionDueTo = OrbitCorrectionReason.Escaping;
+
                             fc.attitude = -o.Prograde(UT);
@@ -496,3 +501,6 @@ void Maneuver()
                         }
-                        else if ((!ongoingOrbitCorrection && (o.ApA >= minSafeAltitude) && (o.altitude >= minSafeAltitude)) || (ongoingOrbitCorrection && descending && o.altitude > minSafeAltitude * 1.1f && o.PeA < minSafeAltitude))
+                        else if (descending && o.PeA < minSafeAltitude && (
+                            (ongoingOrbitCorrectionDueTo == OrbitCorrectionReason.None && o.ApA >= minSafeAltitude && o.altitude >= minSafeAltitude) ||
+                            (ongoingOrbitCorrectionDueTo == OrbitCorrectionReason.PeriapsisLow && o.altitude > minSafeAltitude * 1.1f)
+                        ))
                         {
@@ -505,3 +513,3 @@ void Maneuver()
 
-                            ongoingOrbitCorrection = true;
+                            ongoingOrbitCorrectionDueTo = OrbitCorrectionReason.PeriapsisLow;
                             fc.attitude = deltaV.normalized;
@@ -511,3 +519,2 @@ void Maneuver()
                         {
-                            ongoingOrbitCorrection = true;
                             if (o.ApA < minSafeAltitude * 1.1f)
@@ -517,2 +524,3 @@ void Maneuver()
                                 SetStatus("Correcting Orbit (Apoapsis too low)");
+                                ongoingOrbitCorrectionDueTo = OrbitCorrectionReason.ApoapsisLow;
 
@@ -544,2 +552,3 @@ void Maneuver()
                                 SetStatus("Correcting Orbit (Falling inside atmo)");
+                                ongoingOrbitCorrectionDueTo = OrbitCorrectionReason.FallingInsideAtmosphere;
 
@@ -552,3 +561,3 @@ void Maneuver()
                                 SetStatus("Correcting Orbit (Drifting)");
-                                ongoingOrbitCorrection = false;
+                                ongoingOrbitCorrectionDueTo = OrbitCorrectionReason.None;
                             }
@@ -780,3 +789,3 @@ void UpdateStatus()
             // Prioritize safe orbits over combat outside of weapon range
-            bool fixOrbitNow = hasPropulsion && (CheckOrbitDangerous() || ongoingOrbitCorrection);
+            bool fixOrbitNow = hasPropulsion && (CheckOrbitDangerous() || ongoingOrbitCorrectionDueTo != OrbitCorrectionReason.None);
             bool fixOrbitLater = false;
```

Reply:
We might want to double check the conditions in `CheckOrbitDangerous` and `CheckOrbitUnsafe` to make sure they cover the same conditions as in `Maneuver` with `currentStatusMode = StatusMode.CorrectingOrbit`. There's been a few tweaks there and I'm not sure that they're entirely the same anymore.